route/leastconnection.go

```diff
@@ -83,25 +88,52 @@ func (r *LeastConnection) next() *endpointElem {
 	// more than 1 endpoint
 	// select the least connection endpoint OR
 	// random one within the least connection endpoints
-	randIndices := r.randomize.Perm(total)
 
+	var cur *endpointElem
+	var curIsLocal bool
+	var randIdx int
```

Comment:
It looks like these variables are only used inside the for loop. Can they be moved to be scoped only from inside the loop, to prevent any accidents from them having data from previous iterations?